cellar_rebalancer/src/allocation.rs

```diff
@@ -0,0 +1,157 @@
+        let grpc_client = &mut connections.grpc.unwrap();
+
+        let config = APP.config();
+        let cosmos_gas_price = config.cosmos.gas_price.as_tuple();
+        let allocation = to_allocation(tick_range, pair_id.to_string(), eth_gas_price);
+
+        let name = &config.keys.rebalancer_key;
+        let cosmos_key = config.load_deep_space_key(name.clone());
+        let delegate_cosmos_address = cosmos_key.to_address(&contact.get_prefix()).unwrap();
+
+        let fee = Coin {
+            denom: cosmos_gas_price.1,
+            amount: 0u32.into(),
+        };
+
+        match timeout(Duration::from_secs(100), async {
+            loop {
+                // Waiting for new vote period
+                let res = somm_send::query_commit_period(grpc_client).await;
+                if let Ok(val) = res {
+                    if val.vote_period_start == val.current_height {
+                        break val;
+                    }
+                }
+                sleep(Duration::from_secs(1)).await;
+            }
+        })
+        .await
+        {
+            Ok(val) => {
+                // Sending Pre-commits
+                somm_send::send_precommit(
+                    &contact,
+                    delegate_cosmos_address,
+                    cosmos_key,
+                    fee.clone(),
+                    vec![allocation_precommit(&cosmos_key, &allocation, pair_id.to_string()).await],
+                )
+                .await
+                .unwrap();
+            }
+            Err(e) => {
+                println!("Couldn't Send Precommit {:?}", e);
+            }
+        }
+
+        match timeout(Duration::from_secs(100), async {
+            loop {
+                // Checking Pre-commits for validators
+                let res = somm_send::query_allocation_precommits(grpc_client).await;
+                if let Ok(val) = res {
+                    break val;
+                }
+                sleep(Duration::from_secs(1)).await;
+            }
+        })
+        .await
+        {
+            Ok(val) => {
+                // Sending Commits
+                somm_send::send_allocation(
+                    &contact,
+                    delegate_cosmos_address,
+                    cosmos_key,
+                    fee,
+                    vec![allocation],
+                )
+                .await
+                .unwrap();
+            }
+            Err(e) => {
+                println!("Couldn't Send Commit {:?}", e);
+            }
+        }
+    }
+}
+
+pub fn to_allocation(
+    tick_ranges: Vec<proto::TickRange>,
+    pair_id: String,
+    eth_gas_price: u64,
+) -> proto::Allocation {
+    proto::Allocation {
+        vote: Some(proto::RebalanceVote {
+            cellar: Some(proto::Cellar {
+                id: pair_id,
+                tick_ranges: tick_ranges,
+            }),
+            current_price: eth_gas_price,
+        }),
+        salt: "".to_string(), //TODO: Add salt,
```

Comment:
We still need to implement this. I included a suggestion in the previously merged PR:

https://github.com/PeggyJV/cellar_rebalancer_rs/pull/35#discussion_r770039091